File: Assets/Scripts/LuaScripts/UI/UIAboutPanel.lua
require("Core/System")

local M = class("UIAboutPanel", target)

function M:OnAwake()
    M._close = function()
        self:Close()
    end

    self:Listen(self.btn_close.onClick, M._close)
    self:Listen(self.btn_icon.onClick, M.OpenUrl)
end

function M:OnStart()
    local img_icon = self.btn_icon:GetComponent(typeof(CS.UnityEngine.UI.Image))
    -- TODO 看lua如何支持异步 await
    -- 如果后面要继续使用lua，则需要等csharp层api稳定下来后，新增一个lua专用api
    img_icon.sprite =
        self.AssetLoader:LoadAssetRef("Assets/Arts/Textures/UI/icon_github.png", typeof(CS.UnityEngine.Sprite))

    -- 测试
    -- self.AssetLoader:LoadAssetRef("Assets/Arts/Textures/UI/icon_github.png", typeof(CS.UnityEngine.Sprite))
    -- self.AssetLoader:LoadAsset("Assets/Arts/Textures/UI/icon_github.png", typeof(CS.UnityEngine.Sprite))
end

function M.OpenUrl()
    CS.UnityEngine.Application:OpenURL("https://sarofc.gitee.io")
end

return M

File: Assets/Scripts/HotFix/UI/UIAboutPanel.cs
using Cysharp.Threading.Tasks;
using Saro.UI;
using UnityEngine;
using UnityEngine.UI;

namespace Tetris.UI
{
    [UIWindow((int)EGameUI.UIAboutPanel, "Assets/Res/Prefabs/UI/UIAboutPanel.prefab")]
    public sealed partial class UIAboutPanel : UIWindow
    {
        public UIAboutPanel(string path) : base(path)
        {
        }

        protected override async void Awake()
        {
            base.Awake();

            Listen(btn_Mask.onClick, OnClick_Close);
            Listen(btn_icon.onClick, OnClick_OpenURL);

            btn_icon.GetComponent<Image>().sprite =
                 await AssetLoader.LoadAssetRefAsync<Sprite>("Assets/Arts/Textures/UI/icon_github.png");

            // TODO 大图，可以封装成工具类，加载时转圈，加载完成后再附上去
            //AssetLoader.LoadAssetRefAsync<Sprite>("Assets/Arts/Textures/UI/icon_github.png").ContinueWith((sprite) =>
            //{
            //    BtnIcon.GetComponent<Image>().sprite = sprite;
            //}).Forget();
        }

        private void OnClick_OpenURL()
        {
            Application.OpenURL("https://github.com/Sarofc/com.saro.mgf");
        }

        private void OnClick_Close()
        {
            UIManager.Current.HideWindow(EGameUI.UIAboutPanel);
        }
    }


    public partial class UIAboutPanel
    {
        // =============================================
        // code generate between >>begin and <<end
        // don't modify this scope
        //>>begin
        public Button btn_Mask => Binder.GetRef<Button>("btn_Mask");
        public Button btn_icon => Binder.GetRef<Button>("btn_icon");

        //<<end
        // =============================================
    }
}